sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/firestore/FirestoreV1FnRunQueryTest.java

```diff
@@ -117,65 +119,62 @@ public void resumeFromLastReadValue() throws Exception {
                     .setStartAt(
                         Cursor.newBuilder()
                             .setBefore(false)
-                            .addValues(Value.newBuilder().setStringValue("bar"))))
+                            .addValues(Value.newBuilder().setStringValue("bar"))
+                            .addValues(
+                                Value.newBuilder()
+                                    .setReferenceValue(testData.response2.getDocument().getName()))
+                    )
+                    .addOrderBy(
+                        Order.newBuilder()
+                            .setField(FieldReference.newBuilder().setFieldPath("__name__"))
+                            .setDirection(Direction.ASCENDING))
+            )
             .build();
 
-    List<RunQueryResponse> responses =
-        ImmutableList.of(testData.response1, testData.response2, testData.response3);
-    when(responseStream1.iterator())
-        .thenReturn(
-            new AbstractIterator<RunQueryResponse>() {
-              private int invocationCount = 1;
-
-              @Override
-              protected RunQueryResponse computeNext() {
-                int count = invocationCount++;
-                if (count == 1) {
-                  return responses.get(0);
-                } else if (count == 2) {
-                  return responses.get(1);
-                } else {
-                  throw RETRYABLE_ERROR;
-                }
-              }
-            });
-
-    when(callable.call(testData.request)).thenReturn(responseStream1);
-    doNothing().when(attempt).checkCanRetry(any(), eq(RETRYABLE_ERROR));
-    when(responseStream2.iterator()).thenReturn(ImmutableList.of(responses.get(2)).iterator());
-    when(callable.call(request2)).thenReturn(responseStream2);
-
-    when(stub.runQueryCallable()).thenReturn(callable);
-
-    when(ff.getFirestoreStub(any())).thenReturn(stub);
-    when(ff.getRpcQos(any())).thenReturn(rpcQos);
-    when(rpcQos.newReadAttempt(any())).thenReturn(attempt);
-    when(attempt.awaitSafeToProceed(any())).thenReturn(true);
-
-    ArgumentCaptor<RunQueryResponse> responsesCaptor =
-        ArgumentCaptor.forClass(RunQueryResponse.class);
-
-    doNothing().when(processContext).output(responsesCaptor.capture());
-
-    when(processContext.element()).thenReturn(testData.request);
-
-    RunQueryFn fn = new RunQueryFn(clock, ff, rpcQosOptions);
-
-    runFunction(fn);
+    runQueryRetryTest(testData, expectedRetryRequest);
+  }
 
-    List<RunQueryResponse> allValues = responsesCaptor.getAllValues();
-    assertEquals(responses, allValues);
+  @Test
+  public void resumeFromLastReadValue_nestedOrderBy() throws Exception {
```

Comment:
maybe also add a test for backticked fields, e.g. the query has an inequality on field "\`x.y\`" or has an order by on field "\`x.y\`", and also add a field "x.y" to the returned document in newResponse() to support this test.

Reply:
oh I didn't think about this case (that a back ticked field name is inside a nested map), I don't know what the code is going to do but seems that it's handling it just fine?

Reply:
Implemented escaping logic to handle these cases.